oapi_validate.go

```diff
@@ -223,6 +231,9 @@ func performRequestValidationForErrorHandlerWithOpts(next http.Handler, w http.R
 // of validating a request.
 func validateRequest(r *http.Request, router routers.Router, options *Options) (int, error) {
 
+	r.RequestURI = strings.TrimPrefix(r.RequestURI, options.Prefix)
+	r.URL.Path = strings.TrimPrefix(r.URL.Path, options.Prefix)
```

Comment:
ideally this should be a _copy_ not the existing `r`, so we don't manipulate it before it goes into i.e. the error handler